### ISSUE TITLE
Specification basePath is not respected

### ISSUE THREAD

**Issue by nikhilm** [go-swagger/go-swagger#465]
## Problem statement

Even if the swagger specification has a basePath, the generated client's http transport is not initialized with this basepath.

## Swagger specification

```
swagger: '2.0'
info:
  title: My spec
  version: "0.2.23"
# the domain of the service
host: "localhost:8080"
# array of all schemes that your API supports
schemes:
  - https
  - http
# will be prefixed to all paths
basePath: /v1/myfancy
consumes:
  - application/json
produces:
  - application/json
paths:
  /images:
    get:
      summary: "Get all images."
      description: "Blah"
      responses:
        200:
          description: List of images.
          schema:
            type: string
            name: hello
```

## Steps to reproduce

Run `swagger generate client -f reduced.yml`

The client's http transport initialization is 
```transport := httptransport.New("localhost:8080", "", []string{"http", "https"})```

instead of

```transport := httptransport.New("localhost:8080", "/v1/myfancy", []string{"http", "https"})```

**Comment by casualjim** [comment]
I think you fixed this now, can this be closed?

**Comment by nikhilm** [comment]
Fixed in #466
